fdbserver/workloads/WriteTagThrottling.actor.cpp

```diff
@@ -0,0 +1,283 @@
+		tpsRate = getOption(options, LiteralStringRef("tpsRate"), 300.0);
+		numWritePerTr = getOption(options, LiteralStringRef("numWritePerTr"), 1);
+		numReadPerTr = getOption(options, LiteralStringRef("numReadPerTr"), 1);
+		numClearPerTr = getOption(options, LiteralStringRef("numClearPerTr"), 1);
+
+		writeThrottle = getOption(options, LiteralStringRef("writeThrottle"), false);
+		badActorPerClient = getOption(options, LiteralStringRef("badActorPerClient"), 1);
+		goodActorPerClient = getOption(options, LiteralStringRef("goodActorPerClient"), 1);
+		actorCount = goodActorPerClient + badActorPerClient;
+
+		keyCount = getOption(options, LiteralStringRef("keyCount"),
+		                     std::max(3000, actorCount * 3)); // enough keys to avoid too many conflicts
+		trInterval = actorCount * 1.0 / getOption(options, LiteralStringRef("trPerSecond"), 1000);
+
+		badReadTag = TransactionTag(std::string("badReadTag"));
+		badWriteTag = TransactionTag(std::string("badWriteTag"));
+		goodTag = TransactionTag(std::string("goodTag"));
+	}
+
+	virtual std::string description() { return WriteTagThrottlingWorkload::NAME; }
+
+	// choose a tag
+	ACTOR static Future<Void> _setup(Database cx, WriteTagThrottlingWorkload* self) {
+		state TransactionPriority priority = deterministicRandom()->randomChoice(allTransactionPriorities);
+		state TagSet tagSet1;
+		state TagSet tagSet2;
+		state TagSet tagSet3;
+		tagSet1.addTag(self->badWriteTag);
+		tagSet2.addTag(self->badReadTag);
+		tagSet3.addTag(self->goodTag);
+
+		wait(ThrottleApi::throttleTags(cx, tagSet1, self->tpsRate, self->throttleDuration, TagThrottleType::MANUAL,
+		                               priority));
+		wait(ThrottleApi::throttleTags(cx, tagSet2, self->tpsRate, self->throttleDuration, TagThrottleType::MANUAL,
+		                               priority));
+		wait(ThrottleApi::throttleTags(cx, tagSet3, self->tpsRate, self->throttleDuration, TagThrottleType::MANUAL,
+		                               priority));
+
+		return Void();
+	}
+	Future<Void> setup(const Database& cx) override { return clientId ? Void() : _setup(cx, this); }
+	ACTOR static Future<Void> _start(Database cx, WriteTagThrottlingWorkload* self) {
+		vector<Future<Void>> clientActors;
+		int actorId;
+		for (actorId = 0; actorId < self->goodActorPerClient; ++actorId) {
+			clientActors.push_back(clientActor(false, actorId, 0, cx, self));
+		}
+		for (actorId = 0; actorId < self->badActorPerClient; ++actorId) {
+			clientActors.push_back(clientActor(true, actorId, self->badOpRate, cx, self));
+		}
+		wait(timeout(waitForAll(clientActors), self->testDuration, Void()));
+		return Void();
+	}
+	virtual Future<Void> start(Database const& cx) { return _start(cx, this); }
+	virtual Future<bool> check(Database const& cx) {
+		if (badActorTrNum == 0 && goodActorTrNum == 0) {
+			TraceEvent(SevError, "NoTransactionsCommitted");
+			return false;
+		}
+		if (writeThrottle) {
+			if (!badActorThrottleRetries) {
```

Comment:
This looks good to me. Did you run correctness in your internal Joshua to confirm the test passes? 
You only need to run this new test workload (txt file) for several thousands I guess.

Reply:
Yes, I have run that. It passed 10K test.